plainly-plugin/src/node/collectProject/collectProject.ts

```diff
@@ -26,188 +19,131 @@ function selectFolder(callback: (result: string) => void) {
+  const footageDirRenamed = path.join(aepFileDir, `${randomPrefix}-(Footage)`);
+  const fontsDirRenamed = path.join(aepFileDir, `${randomPrefix}-(Fonts)`);
 
-    output.on('close', () => {
-      console.log(`Zipped ${archive.pointer()} total bytes`);
-      console.log(`Zip file created at: ${zipPath}`);
-      resolve(zipPath);
-    });
+  const undoStack: (() => Promise<void>)[] = [];
 
-    archive.on('error', (err: unknown) => {
-      reject(err);
-      return;
-    });
+  try {
+    await renameIfExists(footageDir, footageDirRenamed);
+    undoStack.push(() => renameIfExists(footageDirRenamed, footageDir));
 
-    archive.pipe(output);
+    await renameIfExists(fontsDir, fontsDirRenamed);
+    undoStack.push(() => renameIfExists(fontsDirRenamed, fontsDir));
 
-    // Append the directory content to the archive
-    archive.directory(targetPathResolved, false);
+    await copyFootage(
+      projectInfo.footage,
+      aepFileDir,
+      footageDir,
+      footageDirRenamed,
+    );
+    undoStack.push(() => removeFolder(footageDir));
 
-    archive.finalize();
-  });
-}
+    await copyFonts(projectInfo.fonts, aepFileDir);
+    undoStack.push(() => removeFolder(fontsDir));
 
-/**
- * Removes the folder at the specified targetPath.
- *
- * @param targetPath The path of the folder to remove.
- */
-async function removeFolder(targetPath: string) {
-  const pathResolved = finalizePath(targetPath); // Normalize and resolve
-
-  try {
-    await fsPromises.rmdir(pathResolved, {
-      maxRetries: 3,
-      recursive: true,
+    await evalScriptAsync('relinkFootage()');
+    undoStack.unshift(async () => {
+      evalScriptAsync('undoFootage()');
     });
-  } catch (error) {
-    console.error(error);
-  }
-}
 
-// Function to make a project zip with no target path specified
-async function makeProjectZipTmpDir() {
-  await fsPromises.mkdir(TMP_DIR, { recursive: true });
-  return makeProjectZip(TMP_DIR);
-}
-
-// Function to make a project zip with a specified target path
-async function makeProjectZip(targetPath: string) {
-  const collectFilesDir = await collectProjectFiles(targetPath);
-  const projectPath = await evalScriptAsync('getProjectPath()');
-  if (!projectPath) {
-    throw new Error('Project not opened or not saved');
+    const zipPath = finalizePath(path.join(targetPath, `${aepFileName}.zip`));
+    await zipItems(zipPath, [
+      { src: aepFilePath, dest: path.basename(aepFilePath), isRequired: true },
+      {
+        src: footageDir,
+        dest: '(Footage)',
+        isRequired: projectInfo.footage.length > 0,
+      },
+      {
+        src: fontsDir,
+        dest: '(Fonts)',
+        isRequired: projectInfo.fonts.length > 0,
+      },
+    ]);
+
+    return zipPath;
+  } finally {
+    for (let i = undoStack.length - 1; i >= 0; i--) {
+      try {
+        await undoStack[i]();
+      } catch (undoErr) {
+        // TODO: What to do here?
+        console.error('Undo failed:', undoErr);
```

Comment:
if something happens here, we need to be sure that we can revert the whole project back... maybe we should zip .aep file before changing it - just in case